components/scream/src/physics/shoc/CMakeLists.txt

```diff
@@ -73,10 +92,16 @@ if (NOT SCREAM_DEBUG)
 endif()
 
 add_library(shoc ${SHOC_SRCS})
-target_include_directories(shoc PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../share)
-set_target_properties(shoc PROPERTIES Fortran_MODULE_DIRECTORY ${SCREAM_F90_MODULES})
-target_link_libraries(shoc physics_share scream_share)
-target_compile_options(shoc PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:${SCREAM_Fortran_FLAGS}>)
+add_library(shoc_sk ${SHOC_SRCS} ${SHOC_SK_SRCS})
```

Comment:
I think we should build the SK lib only if bot SCREAM_SMALL_KERNELS and SCREAM_LIB_ONLY are OFF. If the former is ON, then shoc itself will contain small kernels (right?), while if the latter is ON, then we are not going to run tests anyways...

Reply:
@bartgol , I thought about this too. If SCREAM_SMALL_KERNELS is on, then shoc and shoc_sk are testing the same thing. The shoc_sk test only runs shoc_main_bfb, so I assumed this was an acceptable level of redundancy given that SCREAM_SMALL_KERNELS will usually be off, so why complicate the CMake files even more. If you feel strongly, I will change it to eliminate unneeded builds/tests.

Reply:
I don't feel strongly. We can leave it as is, and perhaps change later if need be.

Reply:
@bartgol , ah, I hit link problems with SMALL_KERNELS on. You are right, this chunk of code needs a refactor.